pallets/rmrk-core/src/functions.rs

```diff
@@ -86,17 +89,18 @@ where
 				thumb.is_none();
 		ensure!(!empty, Error::<T>::EmptyResource);
 
-		let res = ResourceInfo {
-			id: resource_id,
+		let res = ResourceInfo::<BoundedVec<u8, T::ResourceSymbolLimit>, BoundedVec<u8, T::StringLimit>> {
+			id: resource_id.clone(),
 			base,
 			src,
 			metadata,
 			slot,
 			license,
 			thumb,
+			parts,
 			pending: root_owner != sender,
 		};
-		Resources::<T>::insert((collection_id, nft_id, resource_id), res);
+		Resources::<T>::insert((collection_id, nft_id, resource_id.clone()), res);
```

Comment:
I don't think `clone()` is needed here since it wasn't moved at line 93.

Reply:
it was needed because we move it here and then a few lines later we return it.
that said, I changed it so instead we return a blank Ok(()) since we don't really need to pass the resource_id into and then back out of the resource_add function.  fixed in 49db3b20f4c2dd60726c07cb977ef073e1eab9c4